Add plugin: Enhanced Publisher

chararch opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/chararch/obsidian-enhanced-publisher

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[# Obsidian增强发布插件](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[// 使用设置项作为主标题](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/src/settings.ts#L41-L44)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[if (!folder) {](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/src/image-handler.ts#L63)
This check can also return true if there is a file without an extension, this should be a `instanceof TFolder` check.

[await MarkdownRenderer.renderMarkdown(](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/src/html-preview.ts#L551-L556)
This method is deprecated, use `MarkdownRenderer.render` instead, this will also get rid of the console error.

[export async function findAttachmentPath(plugin: EnhancedPublisherPlugin, activeFile: TFile, filename: string): Promise<string | null> {](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/src/html-preview.ts#L638-L738)
Use the `getFirstLinkpathDest` function for this instead.

[function arrayBufferToBase64(buffer: ArrayBuffer): string {](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/src/html-preview.ts#L741-L751)
The Obsidian API already includes this function.

[/* 有图片的文档样式 */](https://github.com/chararch/obsidian-enhanced-publisher/blob/64a569667f0d1b6606cee4b5515d65973385f5f4/src/managers/file-explorer-enhancer.ts#L250-L286)
This part should be in the `styles.css`

chararch commented:
Thank you for the review. I have addressed all the requested changes:

1. Changed the plugin name to avoid using "Obsidian" prefix
2. Removed the top-level heading in settings
3. Updated folder check to use `instanceof TFolder`
4. Replaced deprecated `renderMarkdown` with `MarkdownRenderer.render`
5. Replaced custom `findAttachmentPath` with `getFirstLinkpathDest`
6. Removed custom `arrayBufferToBase64` and used Obsidian's built-in function
7. Moved all styles from `file-explorer-enhancer.ts` to `styles.css`, keeping only the dynamic asset folder hiding styles in the TypeScript file

All changes have been committed and tested. Please let me know if there's anything else that needs to be addressed.
